File: python/Day5/BankOps/Account.py
ops={}
def create_account(name, balance=0):
    ops[name]=balance
    print("Account for", name, "created with balance", balance)

def deposite(account, amount):
    if account in ops:
        ops[account]+=amount
        print(amount, "deposited in", account,"'s account. New balance :", ops[account])
    else:
        print("Account not found")

def withdraw(account, amount):
    if account in ops:
        ops[account]-=amount
        print(amount, "withdrew from", account,"'s account. New balance :", ops[account])
    else:
        print("Account not found")

def checkbalance(account):
    if account in ops:
        print(account,"'s account balance is:", ops[account])
    else:
        print("Account not found")

def transfer_money(sender, receiver, amount):
    if sender in ops and receiver in ops:
        if ops[sender]>=amount:
            ops[sender]-=amount
            ops[receiver]+=amount
            print(amount, "Transferred from", sender, "to", receiver, "'s account")
        else:
            print("Insufficient Funds")
    else:
        print("One or both accounts not found")

create_account("Aman", 10000)
create_account("Arnav", 15000)
deposite("Aman", 5000)
withdraw("Arya", 5000)
withdraw("Arnav", 5000)
checkbalance("Arnav")
transfer_money("Aman", "Arnav", 3000)
checkbalance("Aman")
checkbalance("Arnav")

File: java/Assignments/Day 9/src/Q1/Account.java
package Q1;

public class Account{
    private double balance;
    public Account(double balance){
        this.balance = balance;
    }
//    public double getBalance(){
//        return balance;

    public synchronized void withdraw(int amount) {
        while(this.balance < amount) {
            System.out.println("Insufficient funds. Wait for Deposit.");
            try {
                wait();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        this.balance =  balance - amount;
        System.out.println("Withdrawn Successfully. New Balance: " + this.balance);
        notifyAll();
    }

    public synchronized void deposit(int amount) {
        this.balance =  balance + amount;
        System.out.println("Deposited Successfully. New Balance: " + this.balance);
        notifyAll();
    }
    }
